butane_core/src/codegen/dbobj.rs

```diff
@@ -36,7 +36,14 @@ pub fn impl_dbobject(ast_struct: &ItemStruct, config: &Config) -> TokenStream2 {
     let values: Vec<TokenStream2> = push_values(ast_struct, |_| true);
     let insert_cols = columns(ast_struct, |f| !is_auto(f));
 
-    let save_core = if is_auto(&pk_field) {
+    let save_core = if auto_pk && values.len() == 1 {
```

Comment:
I don't feel strongly enough to block because I don't think it's going to create problems, but I'm not sure I'm personally convinced the changes between here and line 65 are an improvement. You did reduce nesting at line 52, but at the expensive of duplicating lines 41-43, and I'm never a fan of duplicated code.

Reply:
Thanks for reducing the reliance on literals in the generated code though. That part of this change I definitely agree is an improvement.

Reply:
line 42
```rust
let pk = conn.insert_returning_pk(Self::TABLE, &[], &pkcol, &[])?;
```
and line 68
```rust
let pk = conn.insert_returning_pk(Self::TABLE, &[#insert_cols], &pkcol, &values)?;
```
are different.
Hopefully now tables with only a pk (but can include many fields which are virtual columns) will allow the compiler to completely remove the generated `let mut values: Vec....` which is unused in this scenario.  I have one of these in my schema with a few Many's on it.

Reply:
I've now de-dup-ed the `self.#pkident = ...` part also.

Is this better, or worse?

Reply:
Works for me, thanks!